client_authentication.go

```diff
@@ -233,7 +233,7 @@ func (f *Fosite) DefaultClientAuthenticationStrategy(ctx context.Context, r *htt
 		// If this isn't an OpenID Connect client then we actually don't care about any of this, just continue!
 	} else if ok && form.Get("client_id") != "" && form.Get("client_secret") != "" && oidcClient.GetTokenEndpointAuthMethod() != "client_secret_post" {
 		return nil, errorsx.WithStack(ErrInvalidClient.WithHintf("The OAuth 2.0 Client supports client authentication method '%s', but method 'client_secret_post' was requested. You must configure the OAuth 2.0 client's 'token_endpoint_auth_method' value to accept 'client_secret_post'.", oidcClient.GetTokenEndpointAuthMethod()))
-	} else if _, _, basicOk := r.BasicAuth(); basicOk && ok && oidcClient.GetTokenEndpointAuthMethod() != "client_secret_basic" {
+	} else if id, secret, basicOk := r.BasicAuth(); basicOk && ok && id != "" && secret != "" && oidcClient.GetTokenEndpointAuthMethod() != "client_secret_basic" {
```

Comment:
I think we should always assume that `id` is there?

Reply:
Sure, I just followed the `form.Get("client_id") != ""` logic above.  `r.BasicAuth()` will return `"", "", true` if the basic auth header is set to `:` (i.e. empty username and password).

Reply:
Fixed.